src/routes/safe/components/Transactions/TxList/hooks/useLastTxNonce.ts

```diff
@@ -0,0 +1,17 @@
+import { useSelector } from 'react-redux'
+import { queuedTransactions } from 'src/logic/safe/store/selectors/gatewayTransactions'
+
+interface LastTransactionNonce {
+  lastTxNonce: number | undefined
+}
+
+const useLastTxNonce = (): LastTransactionNonce => {
+  const queuedTxs = useSelector(queuedTransactions)
+  const nonces = queuedTxs && Object.keys(queuedTxs)
+
+  const lastTxNonce = nonces && !!nonces?.length ? +nonces[nonces.length - 1] : undefined
```

Comment:
Please try to avoid !! and +.
```suggestion
  const lastTxNonce = nonces?.length ? Number(nonces[nonces.length - 1]) : undefined
```